src/wp-includes/rest-api.php

```diff
@@ -1776,3 +1772,132 @@ function rest_filter_response_by_context( $data, $schema, $context ) {
 
 	return $data;
 }
+
+
+/**
+ * Outputs a header link to a REST resource
+ *
+ * @since 5.5.0
+ *
+ * @param string       $url URL.
+ */
+function rest_head_link( $url ) {
+	if ( ! empty( $url ) ) {
+		printf( '<link rel="alternate" type="application/json" href="%s" />', esc_url( $url ) );
+	}
+}
+
+/**
+ * Return the rest route for a post.
+ *
+ * @since 5.5.0
+ *
+ * @param WP_Post       $post Post Object.
+ * @return string           Route.
+ */
+function rest_get_route_for_post( $post ) {
+	if ( ! $post instanceof WP_Post ) {
+		return '';
+	}
+
+	$post_type = get_post_type_object( $post->post_type );
+	if ( ! $post_type ) {
+		return '';
+	}
+
+	$route      = '';
+	$controller = $post_type->get_rest_controller();
+	if ( ! $controller ) {
+		return '';
+	}
```

Comment:
Let's introduce `$route` only after the last early return:

```suggestion
	$controller = $post_type->get_rest_controller();
	if ( ! $controller ) {
		return '';
	}

	$route = '';
```